### ISSUE TITLE
Spring Boot 2.2.0: ClassPathScanningCandidateComponentProvider findCandidateComponents method

### ISSUE THREAD

**Issue by graiffr** [spring-projects/spring-boot#18684]
After upgrading to version 2.2.0 the method `findCandidateComponents` of class `ClassPathScanningCandidateComponentProvider`, seems not found the classes with the specified user package name.

**Comment by wilkinsona** [comment]
Thanks for the report, but I can't tell what the problem might be from the information that you have provided thus far. `ClassPathScanningCandidateComponentProvider` is part of Spring Framework so it could be that the problem lies there rather than in Spring Boot. Its ability to find entries on the classpath is also affected by things like Spring Boot's `LauchedURLClassLoader` and fat jar support so the problem could be in Spring Boot.

In short, if you'd like us to spend some time trying to diagnose the problem that you are having, please take some time to describe it in detail and to provide a small sample that reproduces it, in the form of a project that has been zipped up and attached to this issue.

**Comment by spring-issuemaster** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by spring-issuemaster** [comment]
Closing due to lack of requested feedback. If you would like us to look at this issue, please provide the requested information and we will re-open the issue.

**Comment by graiffr** [comment]
Hello, I have attached 
[BootTest.zip](https://github.com/spring-projects/spring-boot/files/4152646/BootTest.zip)
a little test code so you can investigate about the issue.

**Comment by wilkinsona** [comment]
Thanks for the sample. There is no retention policy on your custom annotations so it defaults to `RetentionPolicy.CLASS`. If you configure runtime retention (`@Retention(RetentionPolicy.RUNTIME)`) you will see that `BeanTwo` is found as expected.

**Comment by graiffr** [comment]
You are right. Thank you.
